### PR TITLE
fix(web): storage UI fixes and improvements

### PR DESCRIPTION
**Pull request by dgdavid**
## Problem

No long ago a [full UI rewrite](https://github.com/openSUSE/agama/pull/1202) was started and it was recently merged into `master` branch despite not being fully finished. It was just in a good shape for adhering to the [release early, release often](https://en.wikipedia.org/wiki/Release_early,_release_often) philosophy. The idea is to add fixes and improvements little by little against `master` branch according to the (internally) assigned priorities.

But the [openSUSE conference](https://events.opensuse.org/conferences/oSC24) is around the corner and there are few details we'd like to have more polished for the [Agama in action](https://events.opensuse.org/conferences/oSC24/program/proposals/4560) and [The future of Agama and openSUSE](https://events.opensuse.org/conferences/oSC24/program/proposals/4557) events.

## Solution

Address some of these details as part of this PR. Go commit by commit to see them.

## Testing

- Tested manually
- Adapted / re-enabled some unit tests.

## Note for reviewers

No changelog entry required.

## Screenshots

<details>
<summary>Click to show/hide</summary>

---

* Transactional root file system alert

  | Before | After |
  |-|-|
  | Screenshot from 2024-06-17 13-59-15 | Screenshot from 2024-06-17 13-58-36 |
  |Screenshot from 2024-06-17 13-59-19 |Screenshot from 2024-06-17 13-58-46 |

* TPM reminder look&feel

  | Before | After |
  |-|-|
  | Screenshot from 2024-06-17 17-12-26 | Screenshot from 2024-06-17 17-11-28 |

* `Accept` button when manually selecting the boot device

  | Before | After |
  |-|-|
  | Screenshot from 2024-06-17 22-14-04 | Screenshot from 2024-06-17 22-13-02 |

* Size unit selector

  | Before | After |
  |-|-|
   | Screenshot from 2024-06-17 22-41-15  | Screenshot from 2024-06-17 22-39-44 |
</details>

---

Related to https://trello.com/c/H3oZyzXS (internal link)

### CODE REVIEW

File: web/src/components/storage/BootSelection.jsx [hunk 1]
```diff
@@ -88,7 +87,7 @@ export default function BootSelectionDialog() {
 
       setState({
         load: true,
-        bootDevice: availableDevices.find(d => d.name === bootDevice),
+        bootDevice: availableDevices.find(d => d.name === bootDevice) || availableDevices[0],
```
Comment (ancorgs) [inline web/src/components/storage/BootSelection.jsx:90]: I would like @joseivanlopez to take a look to this part. I understood him there is something wrong beyond that simple fallback.

**Review by ancorgs — COMMENTED** [review]
It generally looks good. Just one comment because I suspect the problem with boot disk may not be strictly cosmetic. @joseivanlopez should know more.

File: web/src/components/storage/BootSelection.jsx (hunk 1 above)
Comment (joseivanlopez) [reply web/src/components/storage/BootSelection.jsx:90]:
```suggestion
        bootDevice: availableDevices.find(d => d.name === bootDevice) || defaultBootDevice|| availableDevices[0],
```

Comment (joseivanlopez) [reply web/src/components/storage/BootSelection.jsx:90]: Anyway, I have to do a fix in the rust code. Currently, `defaultBootDevice` is always `undefined`.

Comment (dgdavid) [reply web/src/components/storage/BootSelection.jsx:90]: Sure, if you find a better fix for this in time I'm more than happy to change the introduced hotfix. Think that the BootSelection(Page) component will be re-visited in next iterations, among other components. In any case, I (manually) tested that the patch makes the selection work as expected.

**Review by joseivanlopez — APPROVED** [review]

File: web/src/components/storage/BootSelection.jsx [hunk 2]
```diff
@@ -88,7 +87,7 @@ export default function BootSelectionDialog() {
 
       setState({
         load: true,
-        bootDevice: availableDevices.find(d => d.name === bootDevice),
+        bootDevice: availableDevices.find(d => d.name === bootDevice) || defaultBootDevice || availableDevices[0],
```
Comment (joseivanlopez) [inline web/src/components/storage/BootSelection.jsx:90]: Actually my proposed fix is wrong. Please, revert it. I will create a PR with some fixes for the default boot device.
```suggestion
        bootDevice: availableDevices.find(d => d.name === bootDevice) || availableDevices[0],
```

**Pull request merged**